components/Common/Header/Header.tsx

```diff
@@ -1,25 +1,23 @@
 import styled from "styled-components";
 import Link from "next/link";
 import Image from "next/image";
-import { UserDataInterface } from "@/interfaces";
-import { useRouter } from "next/router";
+import { useIsLogin } from "@/hooks/useIsLogin";
+import { useGetUser } from "@/hooks/useGetUser";
 
 // 컴포넌트의 props 타입 정의의 경우 항상 해줘야한다.
 interface HeaderProps {
-    login: boolean;
-    // ToDo :userData의 경우 sharedPage와 folderPage에서 사용되는데, FolderPageUserDataInterface와 SharedPageUserDataInterface를
-    // 유니온 타입으로 정의했는데 자꾸 에러가 뜨는 이슈 고치기
-    userData?: UserDataInterface;
+    currentPath: string;
```

Comment:
currentPath에 string 타입은 너무 넓은 거 같아요.
코드를 보니까 "shared" "folder" 등 유니언 타입을 선언해서 타입을 좁혀주시는 게 좋습니다
타입은 가능한 좁혀야 type-safe하게 사용하는 방법이에요. 
```suggestion
	currentPath: "shared" | "folder";
```\
위와 같이 하면, 타입 선언되지 않은 값을 currentPath와 비교하려고 할 때, vscode가 잘못된 비교라고 알려줘요~ 아래 스크린샷 참고해보세요
![스크린샷 2024-03-06 오전 12 33 13](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/assets/59413128/c140c527-357b-470f-b937-f4f825ce5b3f)